syserol/regression.py

```diff
@@ -36,52 +36,43 @@ def function_elastic_net(function="ADCC"):
     df_variables = df_merged.drop(["subject"] + functions, axis=1)
 
     # perform regression
-    Y_pred = elasticNetFunc(df_variables, Y)
-
+    enet = elasticNetFunc(df_variables, Y)
+    Y_pred = cross_val_predict(enet, df_variables, Y, cv=len(Y), n_jobs=-1)
     return Y, Y_pred, np.sqrt(r2_score(Y, Y_pred))
 
 
 def function_prediction(tensorFac, function="ADCC", evaluation="all"):
     """ Predict functions using our decomposition and regression methods"""
     func, _ = importFunction()
+    indices = AlterIndices()
 
     Y = func[function]
     X = tensorFac[1][0]  # subjects x components matrix
-    dropped = np.nonzero(np.isnan(Y.to_numpy()))
-    X = X[np.isfinite(Y), :]
+    notAlter = np.delete(np.arange(181), np.unique(np.concatenate((indices, np.where(np.isnan(Y))[0]))))
+    dropped = np.unique(np.concatenate((np.nonzero(np.isnan(Y.to_numpy()))[0], notAlter)))
+    Y_notAlter = Y[dropped][np.isfinite(Y)]
+    X_notAlter = X[Y_notAlter.index]
+    Y_Alter = Y[indices][np.isfinite(Y)]
+    X_Alter = X[Y_Alter.index]
     Y = Y[np.isfinite(Y)]
+    X = X[Y.index]
 
-    Y_pred = elasticNetFunc(X, Y)
+    enet = elasticNetFunc(X_Alter, Y_Alter)
 
     if evaluation == "all":
```

Comment:
Why include "all" if we never use it?

Reply:
All is currently used in fig S1 - is that still necessary?